hermes/bin/src/runtime_extensions/hermes/ipfs/event.rs

```diff
@@ -0,0 +1,29 @@
+//! Hermes IPFS runtime extension event handler implementation.
+use crate::{
+    event::HermesEventPayload,
+    runtime_extensions::bindings::hermes::ipfs::api::{PubsubMessage, PubsubTopic},
+};
+
+/// Event handler for the `on-topic` event.
+#[allow(dead_code)]
+pub(crate) struct OnTopicEvent {
+    ///  Topic
+    pub(crate) topic: PubsubTopic,
+    ///  Message
+    pub(crate) message: PubsubMessage,
+}
+
+impl HermesEventPayload for OnTopicEvent {
+    fn event_name(&self) -> &str {
+        "on-topic"
+    }
+
+    fn execute(&self, module: &mut crate::wasm::module::ModuleInstance) -> anyhow::Result<()> {
+        let _res: bool = module
+            .instance
+            .hermes_ipfs_event()
+            .call_on_topic(&mut module.store, &self.message)?;
```

Comment:
Do we need to send `topic` data as well ?

Reply:
Yes, thanks for catching this, I forgot to update `pubsub-message` to be a record instead of a simple type.